src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -1110,11 +1110,11 @@ private void LoadExtensions(Assembly assembly, bool throwOnError)
         /// </remarks>
```

Comment:
I am not entirely convinced this is a valid test. I don't see any reason why some of the ML.NET product assemblies can't contain extensions. The HasMLNetPublicKey test is also no longer valid because product and shipping assemblies are now signed with the same key. 
Can we delete the related code?

Reply:
I left it in because I wasn't sure if we wanted to try and find a way to fix the code I commented out or not. If you don't think it should be fixed then I can go ahead and remove all the associated code.

Reply:
I can see a scenario where we choose to include a bunch of easy data transforms using custom transformers (like the normalizers we had discussed before). I think it is okay to remove it.

@eerhardt Any thoughts on if we this needs to be retained?


---
In reply to: [513676716](https://github.com/dotnet/machinelearning/pull/5464#discussion_r513676716) [](ancestors = 513676716)

Reply:
See the comment on the only callsite to this method:

https://github.com/dotnet/machinelearning/blob/d849ba4c7831586018d5b2cca6bdd2b1fa228ddf/src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs#L1077-L1079

Reply:
In Arcade, we are signing product and test binaries with the same key. Is there any other way to distinguish product binaries from test binaries? If not, do we lose anything from scanning product binaries as well (apart from a small perf hit)?




---
In reply to: [513810237](https://github.com/dotnet/machinelearning/pull/5464#discussion_r513810237) [](ancestors = 513810237)

Reply:
> Is there any other way to distinguish product binaries from test binaries?

The check also checks the name of the assembly starts with `Microsoft.ML`. Another option would be to change the strong name key for the single test assembly that needs this. (It doesn't really matter which strong name we give our test assemblies)

> apart from a small perf hit

I have not measured it, so I'm not certain I would say it is "small". It affects startup. Our assemblies contain a lot of types, which we know do not contain this attribute. So it is 100% wasteful to do this scanning.

Reply:
@michaelgsharp Can you please try if changing the strong name key works?

---
In reply to: [513823854](https://github.com/dotnet/machinelearning/pull/5464#discussion_r513823854) [](ancestors = 513823854)

Reply:
Fixed the checked so it excludes Our test projects.

Reply:
Can you please add a note here about the change. Something along the lines that we used to have two separate strong name keys for product and test binaries and Arcade does not support that anymore (and that did not accept our PR for adding the additional signed key and using two strong name keys without the PR) ?

---
In reply to: [516842984](https://github.com/dotnet/machinelearning/pull/5464#discussion_r516842984) [](ancestors = 516842984)

Reply:
Comment added.